io/net/ethtool_test.py

```diff
@@ -102,19 +117,40 @@ def test_ethtool(self):
                               ignore_status=True)
             if ret.exit_status != 0:
                 self.fail("failed")
-        if not self.ping_check('-c 5'):
-            self.fail("ping failed after interface is up")
+        if not HostInfo.ping_check(self, self.iface, self.peer, '10000',
+                                   flood=True):
+            self.fail("flood ping test failed")
+        if self.priv_test:
+            self.ethtool_toggle_priv_flags()
 
-    def ping_check(self, options):
+    def ethtool_toggle_priv_flags(self):
         '''
-        Checks if ping to peer works fine and returns True.
-        Returns False otherwise.
+        Toggle the priv flag settings of the driver.
         '''
-        cmd = "ping -I %s %s %s" % (self.iface, options, self.peer)
-        if process.system(cmd, shell=True, verbose=True,
-                          ignore_status=True) != 0:
-            return False
-        return True
+        priv_pass = []
+        priv_fail = []
+        for oper in ('toggle', 'setback'):
+            for line in self.ret_val.stdout_text.splitlines():
+                if "off" in line:
+                    val = "on"
+                else:
+                    val = "off"
+                if "flags" not in line:
+                    priv_flag = line.split(':')[0]
+                    cmd = "ethtool --set-priv-flags %s \"%s\" %s" % \
+                          (self.iface, priv_flag.rstrip(), val)
+                    ret1 = process.run(cmd, shell=True, verbose=True,
+                                       ignore_status=True)
+                    if ret1.exit_status == 0 or 'supported' in \
+                       ret1.stderr_text:
+                        priv_pass.append(priv_flag.rstrip())
+                    else:
+                        priv_fail.append(priv_flag.rstrip())
+            if not HostInfo.ping_check(self, self.iface,
+                                       self.peer, '500000', flood=True):
+                self.fail("Ping failed for priv flag oper = %s" % oper)
+        if priv_fail:
+            self.fail("Private flags could not be toggled %s" % set(priv_fail))
```

Comment:
`% " ".join(list(set(priv_fail)))`

Reply:
@narasimhan-v  actually using this method is not giving the desired formated output
submitted patch results for failure condition
:
 (01/16) ethtool_test.py:Ethtool.test_ethtool;PrivateFlag-18f0: FAIL: Private flags could not be toggled {'dropless_rq', 'xdp_tx_mpwqe'} (105.45 s)


newcode:
% " ".join(list(set(priv_fail)))

results:
(01/16) ethtool_test.py:Ethtool.test_ethtool;PrivateFlag-18f0: FAIL: Private flags could not be toggled xdp_tx_mpwqe dropless_rq (103.09 s)

Reply:
with the current patch it is easier to pick the failing private flags

Reply:
@manvanthar I understand, may be you can do something like
``self.fail("Private flags could not be toggled:  %s"`% ",".join(list(set(priv_fail))))```
I am asking to do this since other tests have fail message in this format, and also list should be formatted to string properly.

Reply:
ok @narasimhan-v  done the latest ammend should have the review comments.

Failed output now:

(01/16) ethtool_test.py:Ethtool.test_ethtool;PrivateFlag-18f0: FAIL: Private flags could not be toggled: dropless_rq,xdp_tx_mpwqe (62.08 s)